server/etcdmain/etcd.go

```diff
@@ -90,6 +90,10 @@ func startEtcdOrProxyV2(args []string) {
 		lg.Info("failed to detect default host", zap.Error(dhErr))
 	}
 
+	if cfg.ec.Name == embed.DefaultName {
+		lg.Warn("the server is using default name, which might cause error that failed the startup.")
+	}
```

Comment:
I suggest to add the check into [(*Config) Validate()](https://github.com/etcd-io/etcd/blob/9bc4a63a419cd316f6920e37ec551b87d1c7c3a7/server/embed/config.go#L665), so that we have all configuration checker in one place.

Reply:
Thanks! 
I tried to find such a place, but failed. 
So I put it here according to the content in the log `'data-dir' was empty; using default`...

Reply:
updated.